functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,357 @@
+package createsetters
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
```

Comment:
is there any duplicate code between apply-setters and this function that should b extracted into a common package?

Reply:
apply-setters, create-setters, search-replace have similar `walk.go` files where the visitor interface is defined and recursion calls are made to visit each field.

Reply:
There is an issue for tracking https://github.com/GoogleContainerTools/kpt/issues/2043. @NagaHarshita can work on this issue across the board after merging this PR.